tests/test_sim_results/electric_blue/test_volume_timeseries.py

```diff
@@ -0,0 +1,203 @@
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("month"),
+                            1
+                        ),
+                        "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("month"),
+                            1
+                        )
+                    },
+                    "WEEK_FORWARD": {
+                        "bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("week"),
+                            1
+                        ),
+                        "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("week"),
+                            1
+                        )},
+                    "DAY_FORWARD": {"bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("day"),
+                            1
+                        ), "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            DateTime(2020, 1, 1, 0, 0, 0).end_of("day"),
+                            1
+                        )}}},
+            DateTime(2019, 1, 1, 0, 0, 0): {
+                DateTime(2019, 1, 1, 0, 0, 0): {
+                    "SSP": 36100.3496802018,
+                    "YEAR_FORWARD": {"bought_kWh": 0, "sold_kWh": 0},
+                    "MONTH_FORWARD": {"bought_kWh": 0, "sold_kWh": 0},
+                    "WEEK_FORWARD": {
+                        "bought_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0).start_of("week"),
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            1
+                        ),
+                        "sold_kWh": sum_of_energy(
+                            DateTime(2020, 1, 1, 0, 0, 0).start_of("week"),
+                            DateTime(2020, 1, 1, 0, 0, 0),
+                            1
+                        )
+                    },
+                    "DAY_FORWARD": {"bought_kWh": 0, "sold_kWh": 0}}}}
+
+    def test_add_asset_time_series_for_1_month_resolution(self, forward_time_series):
+        """Call AssetVolumeTimeSeries in 1-month resolution to assure no KeyErrors happen."""
+        volume_time_series = AssetVolumeTimeSeries(
+            asset_uuid="UUID", asset_peak_kWh=5, resolution=AggregationResolution.RES_1_MONTH)
+
+        try:
+            for market_type, asset_stats in forward_time_series:
+                volume_time_series.add(asset_stats, market_type)
+        except KeyError as exc:
+            pytest.fail(str(exc))
+
+        self.check_time_slots_year(volume_time_series)
```

Comment:
Minor, but maybe valuable would be to add a more detailed check here. E.g. an easy test could be to check the number of time steps present in the nested dict. For example the number of time stamps for the monthly resolution should be 12 (and equivalent in the other resolutions too. What do you think?

Reply:
Actually this is done [here](https://github.com/gridsingularity/gsy-framework/blob/c79701368431749b81f8b90d342da71a3ec3a083/tests/forward_markets/test_aggregated_profile.py). But as you just mentioned, more assertions could be added to make it safer. Let me know if you had any other ideas on this. Thanks

Reply:
I see, you mean if the `*AggregatedSSPProfile.generate()` is broken, it will be cought by the tests you linked and does not have to be tested here again, because `AssetVolumeTimeSeries` is calling it. 
LGTM